### PR TITLE
spec: types in schema description

### LINKED ISSUE #162: spec: explicitly define field types in schema definitions
Scroll past the example of a [schema](https://github.com/appc/spec/blob/master/SPEC.md#image-manifest-schema), to where each field is given a description. Some descriptions include a description of the type of the value for the field, while others do not. I propose that things would be more clear if we did something like:

* **acKind** (string, required) - description
* **annotations** (list of objects, optional) - description

This makes it much more clear and doesn't require looking at an example while reading the descriptions of each field. If you agree, I can make a PR.

### PR DESCRIPTION
**Pull request by jzelinskie**
Fixes #162

I also found some mistakes in the example schema while running over this.

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -504,30 +504,30 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
...
-* **labels** (optional) list of label objects (where the *name* is restricted to the AC Name formatting and *value* is an arbitrary string). Label names must be unique within the list, and (to avoid confusion with the image's name) cannot be "name". Labels are used during image discovery and dependency resolution. Several well-known labels are defined:
+* **acKind** (string, required) must be set to "ImageManifest"
+* **acVersion** (string, required) represents the version of the schema specification that the manifest implements (string, must be in [semver](http://semver.org/) format)
+* **name** (string, required) is used as a human readable index to the container image. (string, restricted to the AC Name formatting)
+* **labels** (list of objects, optional) are used during image discovery and dependency resolution. The listed objects should have two key-value pairs: *name* is restricted to the AC Name formatting and *value* is an arbitrary string. Label names must be unique within the list, and (to avoid confusion with the image's name) cannot be "name". Several well-known labels are defined:
```
Comment (cdaylward) [inline SPEC.md]: "two key-value pairs"
The listed objects should have two key-value pairs -> List objects are key-value pairs

File: SPEC.md [hunk 2]
```diff
@@ -504,30 +504,30 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
...
-* **annotations** (optional) list of annotation objects (where the *name* is restricted to the [AC Name](#ac-name-type) formatting and *value* is an arbitrary string). Annotation names must be unique within the list. Annotations can be used by systems outside of the ACE (ACE can override). If you are defining new annotations, please consider submitting them to the specification. If you intend for your field to remain special to your application please be a good citizen and prefix an appropriate namespace to your key names. Recognized annotations include:
+    * **workingDirectory** (string, optional) is the working directory of the launched application, relative to the application image's root (must be an absolute path, defaults to "/", ACE can override). If the directory does not exist in the application's assembled rootfs (including any dependent images and mounted volumes), the ACE must fail execution.
```
Comment (cdaylward) [inline SPEC.md]: The convention in the schema has been to move toward the zero article. Remove "is the"

Comment (cdaylward) [inline SPEC.md]: Same, remove definite article, please.

File: SPEC.md [hunk 3]
```diff
@@ -504,30 +504,30 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
...
+    * **environment** (list of objects, optional) represents the app's environment variables (ACE can append). The listed objects should have two key-value pairs: **name** and **value**. The **name** must consist solely of letters, digits, and underscores '_' as outlined in [IEEE Std 1003.1-2001](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html) and **value** is an arbitrary string.
```
Comment (cdaylward) [inline SPEC.md]: Same as above. They are key/value pairs, not two pairs. Also, please remove "should".
"and **value**" should be a new sentence. The **value** is...

File: SPEC.md [hunk 4]
```diff
@@ -594,7 +596,8 @@ JSON Schema for the Container Runtime Manifest (container manifest), conforming
             "name": "work",
             "kind": "host",
             "source": "/opt/tenant1/work",
-            "readOnly": true
+            "readOnly": true,
+            "fulfills" "/work"
```
Comment (cdaylward) [inline SPEC.md]: fulfills was removed. Looks like the definition below did not get removed in a previous change.

File: SPEC.md [hunk 5]
```diff
@@ -612,22 +615,22 @@ JSON Schema for the Container Runtime Manifest (container manifest), conforming
...
+    * **kind** (string, required) either "empty" or "host". "empty" fulfills a mount point by ensuring the path exists (writes go to container). "host" fulfills a mount point with a bind mount from a **source**.
+    * **source** (string, required if **kind** is "host") absolute path on host to be bind mounted into the container under a mount point.
+    * **readOnly** (boolean, optional if **kind** is "host") whether or not the volume should be mounted read only.
+    * **fulfills** (string, required) MountPoints of the containers that this volume can fulfill (restricted to AC Name formatting)
+* **isolators** (list of objects, optional) list of isolators that will apply to all apps in this container. Each object has two key value pairs: **name** is restricted to the AC Name formatting and **value** can be a freeform string)
+* **annotations** (list of objects, optional) arbitrary metadata the executor should make available to applications via the metadata service. Objects should contain two key-value pairs: **name** is restricted to the [AC Name](#ac-name-type) formatting and **value** is an arbitrary string). Annotation names must be unique within the list.
```
Comment (cdaylward) [inline SPEC.md]: should -> must

**Comment by philips** [comment]
@jzelinskie Ping

**Comment by jzelinskie** [comment]
sorry for the delay @philips

**Comment by jonboulle** [comment]
Ping, needs rebase, and also I still see a few instances that didn't move to the zero article?

**Comment by jzelinskie** [comment]
will address tomorrow morning

**Comment by jzelinskie** [comment]
rebased and moved towards the zero article
@jonboulle @philips

**Comment by philips** [comment]
lgtm

**Comment by jonboulle** [comment]
hurrr, there are still non-zero articles in there!

**Comment by jzelinskie** [comment]
>_< rebasing on prose is hard

**Pull request merged**